### PR TITLE
codeintel: Retry connecting to upload store on worker startup

### PR DESCRIPTION
**Pull request by efritz**
Retry provisioning minio bucket on worker startup when it's a network failure.

### CODE REVIEW

**Review by arussellsaw — APPROVED** [review]

**Comment by arussellsaw** [comment]
thanks! 🙏

File: enterprise/cmd/precise-code-intel-worker/main.go [hunk 1]
```diff
@@ -142,3 +148,31 @@ func mustRegisterQueueMetric(observationContext *observation.Context, dbStore *s
...
+
+func initializeUploadStore(ctx context.Context, uploadStore uploadstore.Store) error {
+	for {
+		if err := uploadStore.Init(ctx); err == nil || !isRequestError(err) {
+			return err
+		}
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```
Comment (mrnugget) [inline enterprise/cmd/precise-code-intel-worker/main.go:160]: This is never triggered though, right? You're passing in `context.Background()`.

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/main.go:160]: For now but I want the extension point there so we don't have a context-unbound infinite loop.

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/main.go:160]: (It's a way to future-proof avoiding needing to cancel the loop in the future but not having a way to signal it without threading in new values.)

**Pull request merged**